src/vivarium_testing_utils/automated_validation/interface.py

```diff
@@ -13,71 +15,79 @@
 
 class ValidationContext:
     def __init__(self, results_dir: str | Path, age_groups: pd.DataFrame | None):
-        self._data_loader = DataLoader(results_dir)
-        self.comparisons = {}
+        self._data_loader = DataLoader(Path(results_dir))
+        self.comparisons: dict[str, Comparison] = {}
 
     def get_sim_outputs(self) -> list[str]:
         """Get a list of the datasets available in the given simulation output directory."""
-        return self._data_loader.sim_outputs()
+        return self._data_loader.get_sim_outputs()
 
     def get_artifact_keys(self) -> list[str]:
         """Get a list of the artifact keys available to compare against."""
-        return self._data_loader.artifact_keys()
+        return self._data_loader.get_artifact_keys()
 
     def get_raw_dataset(self, dataset_key: str, source: str) -> pd.DataFrame:
         """Return a copy of the dataset for manual inspection."""
         return self._data_loader.get_dataset(dataset_key, DataSource.from_str(source))
 
-    def upload_custom_data(self, dataset_key: str, data: pd.DataFrame | pd.Series) -> None:
+    def upload_custom_data(
+        self, dataset_key: str, data: pd.DataFrame | pd.Series[float]
+    ) -> None:
         """Upload a custom DataFrame or Series to the context given by a dataset key."""
+        if isinstance(data, pd.Series):
+            data = data.to_frame(name="value")
         self._data_loader.upload_custom_data(dataset_key, data)
 
     def add_comparison(
-        self, measure_key: str, test_source: str, ref_source: str, stratifications: list[str]
+        self,
+        measure_key: str,
+        test_source: str,
+        ref_source: str,
+        stratifications: list[str] = [],
     ) -> None:
         """Add a comparison to the context given a measure key and data sources."""
-        entity_type, entity, measure = measure_key.split(".")
-        measure = MEASURE_KEY_MAPPINGS[entity_type][measure](entity)
+        entity_type, entity, measure_name = measure_key.split(".")
+        measure = MEASURE_KEY_MAPPINGS[entity_type][measure_name](entity)
 
-        test_source = DataSource.from_str(test_source)
+        test_source_enum = DataSource.from_str(test_source)
 
-        if not test_source == DataSource.SIM:
+        if not test_source_enum == DataSource.SIM:
             raise NotImplementedError(
                 f"Fuzzy Comparison for {test_source} source not implemented. Must be SIM."
```

Comment:
Nit: should this method be called `add_fuzzy_comparison()`, since it's actually specifically creating a `FuzzyComparison`? If not, this error message shouldn't reference fuzzy imo.